compiler/src/iree/compiler/InputConversion/MHLO/ConvertCollectiveOps.cpp

```diff
@@ -129,6 +129,36 @@ static IREE::Flow::CollectiveElementTypeAttr getCollectiveElementTypeAttr(
                                                     *collectiveElemType);
 }
 
+template <typename T>
+static LogicalResult checkCollectiveAttrs(T op, PatternRewriter &rewriter) {
+  // Check there is only one group in the replica_groups
+  ShapedType replicaGroupType = op.getReplicaGroups().getType();
+  if (replicaGroupType.getRank() != 2 || replicaGroupType.getDimSize(0) != 1) {
+    return rewriter.notifyMatchFailure(op, "must have a single replica group");
+  }
+
+  int64_t channelHandle =
```

Comment:
Could we add a helper for this?

Reply:
Do you mean in the op spec? Or in the file as a static function?

The first case, I hope it is done by the MHLO team. For the latter case, the check function is already a helper function, so I don't know if creating another helper used only once helps much.